### PR TITLE
Molecule test

### PR DESCRIPTION
**Pull request by Infinity-ops**
- Added 'molecule_workflow.yml' which triggers the molecule workflow actions to test all the ansible roles in this Repository.
- For now, Molecule is added only for a few existing ansible roles: miniconda, proxies, Poetry in order to verify their installation and their functionality on containers.
- By default, molecule uses docker as a driver for executing provision testing instances for the ansible roles. Further working on how  to overcome  limitations  of docker and github action in implementing the molecule test for other roles.

### CODE REVIEW

**Review by enasca — CHANGES_REQUESTED** [review]
We need to execute tests on Ubuntu 22.04 as well, but other than that it looks ok. Any comments, @hfrSchmidt?

File: ansible/roles/poetry/tasks/poetry-install.yml [hunk 1]
```diff
@@ -45,6 +45,25 @@
...
+
+- name: Run Bash completions command
+  ansible.builtin.command: "{{ poetry_home | quote }}/bin/poetry completions bash"
+  register: "cmd_poetry_bash_completions"
+  changed_when: false
   when: poetry_verification_result is success
+
+- name: Save the Bash completions output to a file
+  ansible.builtin.copy:
+    content: "{{ cmd_poetry_bash_completions }}"
```
Comment (enasca) [inline ansible/roles/poetry/tasks/poetry-install.yml:65]: Shouldn't this be `cmd_poetry_bash_completions['stdout']`?

**Review by enasca — CHANGES_REQUESTED** [review]

**Comment by enasca** [comment]
By the way, those lint errors come from the new version of the linter catching more things. Fixing those now.

**Comment by enasca** [comment]
Fixed the lint errors in #64. After it gets merged, we'll need to rebase this branch.

File: .github/workflows/molecule_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+          
+      - name: Set up Python 
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
```
Comment (enasca) [inline .github/workflows/molecule_workflow.yml:22]: Is it possible to change this back to `3.x`? I don't think we need 3.10 specifically.

**Review by enasca — CHANGES_REQUESTED** [review]

**Review by enasca — APPROVED** [review]

File: ansible/roles/proxies/molecule/default/verify.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+- name: Verify
+  hosts: all
+  become: true
+  tasks:
+  - name: Print variables from proxies.sh
+    command: cat /etc/profile.d/proxy.sh
+    register: "output"
+  - debug: var=output.stdout_lines
```
[the remark this answers is not in the document]
Comment (hfrSchmidt) [reply ansible/roles/proxies/molecule/default/verify.yml:9]: If its not clear how to exactly do this, I suggest we defer the test of the proxies role to the next revision and remove it from the PR for now.

**Review by hfrSchmidt — COMMENTED** [review]
I think it is not sufficient to just test with --version if the software is there in the verify step. This does not really ensure correct installation. 
But it is a start and I think for now it is good enough.

File: ansible/roles/proxies/molecule/default/verify.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (enasca) [reply ansible/roles/proxies/molecule/default/verify.yml:9]: If we invoke the `shell` module in ansible, we should be able to do `echo $HTTPS_PROXY` and verify the output.

[the remark this answers is not in the document]
Comment (hfrSchmidt) [reply ansible/roles/proxies/molecule/default/verify.yml:9]: Yes and then cross check with the values that are supposed to be there?

**Review by enasca — APPROVED** [review]
We're now sourcing the shell script in order to verify if it sets the variables correctly, which looks reasonable to me.

**Review by hfrSchmidt — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**